src/commands/ingress/IngressUpdateBaseStep.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Ingress } from "@azure/arm-appcontainers";
+import { AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { Progress } from "vscode";
+import { ext } from "../../extensionVariables";
+import { IContainerAppContext } from "../IContainerAppContext";
+import { updateContainerApp } from "../deployContainerApp/updateContainerApp";
+
+type IngressOptions = {
+    ingress: Ingress | null,
+    working: string,
+    workCompleted: string
+}
+
+export abstract class IngressUpdateBaseStep<T extends IContainerAppContext> extends AzureWizardExecuteStep<T> {
```

Comment:
This step is meant to eventually completely replace the use of `updateIngressSettings.ts`